elasticapm/transport/base.py

```diff
@@ -59,9 +59,10 @@ def __init__(
 
     def queue(self, event_type, data, flush=False):
         with self._queue_lock:
-            self.queued_data.write((self._json_serializer({event_type: data}) + "\n").encode("utf-8"))
-        since_last_flush = timeit.default_timer() - self._last_flush
-        queue_size = self.queued_data_size
+            queued_data = self.queued_data
+            queued_data.write((self._json_serializer({event_type: data}) + "\n").encode("utf-8"))
+            since_last_flush = timeit.default_timer() - self._last_flush
+            queue_size = 0 if queued_data is None or queued_data.fileobj is None else queued_data.fileobj.tell()
```

Comment:
shouldn't you still be checking if queued_data _has_ a fileobj? i.e. for the case where compression is disabled, and queued_data is a BytesIO?

Reply:
ugh, right. Having the option to turn off the use of gzip compression was probably a mistake in hindsight. Especially because using `compression_level=0` already turns off any compression :/ this might be as good a time as any to remove it. I don't think it would violate backwards compatibility

Reply:
Seems reasonable. Maybe just treat None the same as 0 for compress_level: create a GzipFile with compresslevel=0?

Reply:
done in a479fc7 👍